sdkutil/event.go

```diff
@@ -87,3 +104,172 @@ func GetString(attrs []sdk.Attribute, key string) (string, error) {
 	}
 	return "", ErrNotFound
 }
+
+// BlockID provides type to properly format print blockchain ID types.
+type BlockID struct {
+	Owner    *sdk.AccAddress
+	DSeq     *uint64
+	GSeq     *uint32
+	OSeq     *uint32
+	Provider *sdk.AccAddress
+}
+
+// NewBlockID initializes the BlockID struct, passing nil values is acceptable
+// for fields which are not used.
+func NewBlockID(owner *sdk.AccAddress, dseq *uint64, gseq *uint32, oseq *uint32, provider *sdk.AccAddress) BlockID {
+	return BlockID{
+		Owner:    owner,
+		DSeq:     dseq,
+		GSeq:     gseq,
+		OSeq:     oseq,
+		Provider: provider,
+	}
+}
+
+// String method provides the full formating of all the BlockID fields which are not nil.
+// Format: [Owner AccAddress]/[DSeq]/[GSeq]/[OSeq]/[Provider AccAddress]
+// eg: akash1vgv30hr8lel8r7270wywmak5c82es3njws3u4z/795423625/1/1/akash1pxqksfr60zc6uadht6300uavwl3s58ct0apt9w
+func (b BlockID) String() string {
+	parts := make([]string, 0)
+	if b.Owner != nil {
+		parts = append(parts, b.Owner.String())
+	}
+	if b.DSeq != nil {
+		parts = append(parts, strconv.FormatUint(*b.DSeq, 10))
+	}
```

Comment:
if `b.Owner == nil` then `b.DSeq` should be nil, etc...

:thinking: probably not important to enforce here tho.